app/service/actions.js

```
@@ -30,59 +31,73 @@ export const actions = {
       store.commit('setErrorMessage', error.message);
     });
   },
   navigate: (store) => (folder) => {
     store.commit('setFilesLoading', true);
-    store.serviceFactory.getFileServiceInstance().getFiles(folder)
+    return store.serviceFactory.getFileServiceInstance().getFiles(folder)
         .then((files) => {
           store.commit('setCurrentFolder', folder);
           store.commit('setFiles', files);
           store.commit('setFilesLoading', false);
         }).catch((error) => {
           store.commit('setFilesLoading', false);
           store.commit('setErrorMessage', error.message);
+        }).finally(() => {
+          store.commit('setFilesLoading', false);
         });
   },
   moveFile: (store) => (fileId, folderId) => {
     store.commit('setFilesMoving', true);
-    store.serviceFactory.getFileServiceInstance().moveFile(fileId, folderId).then(() => {
-      store.commit('setFilesMoving', false);
-      store.dispatch('fetchFiles');
-    }).catch((error) => {
-      store.commit('setFilesMoving', false);
-      store.commit('setErrorMessage', error.message);
-    });
+    return store.serviceFactory.getFileServiceInstance().moveFile(fileId, folderId)
+        .then(() => {
+          store.commit('setFilesMoving', false);
+          return store.dispatch('fetchFiles');
+        }).catch((error) => {
+          store.commit('setFilesMoving', false);
+          store.commit('setErrorMessage', error.message);
+        });
+  },
+  createFolder: (store) => (folderName, folder) => {
+    store.commit('setFolderCreating', true);
+    return store.serviceFactory.getFileServiceInstance().createFolder(folderName, folder)
+        .then(() => {
+          store.commit('setFolderCreating', false);
+          return store.dispatch('fetchFiles');
+        })
+        .catch((error) => {
+          store.commit('setFolderCreating', false);
+          store.commit('setErrorMessage', error.message);
+          return error;
```

Should throw this error if you want to receive it in catch().